### PR TITLE
Updates to README for Beta release

### LINKED ISSUE #2857: Update the README goals and scope for our first Beta release
Similar to #2372, when we finish the mempool work and are ready to tag our first beta release, we should update the README.

In order to do this, we will need to explicitly outline any full node functionality which is not yet implemented in Zebra, in particular:

- Any zcash consensus rules not yet implemented in zebra
    - Any sections in the zcash spec that we have not yet implemented
    - Any ZIPs which are not implemented
    - Any bitcoin rules that are not implemented
- Anything else that zcashd does that zebra does not yet do? 
    - RPCs
    - wallet interfaces and wallet functionality
    - mining
    - network protocol bloom filter messages
    - Tor integration

### PR DESCRIPTION
**Pull request by mpguerra**
## Motivation

Closes #2857 

## Review

PR #2894 also updates the README and may conflict with this PR.

### Reviewer Checklist

  - [ ] Code implements Specs and Designs
  - [ ] Tests for Expected Behaviour
  - [ ] Tests for Errors

### CODE REVIEW

**Review by mpguerra — COMMENTED** [review]
Do we want to make any changes or updates to the "Current Features section"?

**Review by upbqdn — DISMISSED** [review]
Looks good to me.

**Comment by conradoplg** [comment]
> Do we want to make any changes or updates to the "Current Features section"?

I think so. My suggestion:

```
Network:
- synchronize the chain from peers
- maintain a transaction mempool
- download gossiped blocks and transactions from peers
- answer inbound peer requests for hashes, headers, blocks and transactions
```

And then in `Future Work`, remove "transaction mempool".

(I can make the changes if needed)

File: README.md [hunk 1]
```diff
@@ -13,17 +13,50 @@ consensus-compatible implementation of the Zcash protocol, currently under
...
+#### NU1 - Sapling
+- Validation of Sapling anchors
+- Validation of sapling note commitment trees
+
+#### NU0 - Overwinter
+- ZIP-203: Transaction Expiry consensus rules
+
+#### Sprout
+- Validation of Sprout anchors
+- Validation of JoinSplit proofs using Groth16 verifier
```
Comment (dconnolly) [inline README.md]: So I think Sprout-on-Groth16 came in a later NU, it might be NU1 (Sapling), just to be precise

Comment (mpguerra) [reply README.md]: Fixed in 464c74e9

**Comment by mpguerra** [comment]
> > Do we want to make any changes or updates to the "Current Features section"?
> 
> I think so. My suggestion:
> 
> ```
> Network:
> - synchronize the chain from peers
> - maintain a transaction mempool
> - download gossiped blocks and transactions from peers
> - answer inbound peer requests for hashes, headers, blocks and transactions
> ```
> 
> And then in `Future Work`, remove "transaction mempool".
> 
> (I can make the changes if needed)

Changes made in 61a8a36 and 4d163df

File: README.md [hunk 2]
```diff
@@ -139,26 +174,30 @@ Zebra primarily depends on pure Rust crates, and some Rust/C++ crates:
 ### Known Issues
 
 There are a few bugs in Zebra that we're still working on fixing:
+- [When Zebra receives an unexpected network message from a peer, it disconnects from that peer #2107](https://github.com/ZcashFoundation/zebra/issues/2107)
+- [A Zebra instance could be used to pollute the peer addresses of other nodes #1889](https://github.com/ZcashFoundation/zebra/issues/1889)
+- [Zebra's address book can use all available memory #1873](https://github.com/ZcashFoundation/zebra/issues/1873)
+- [Zebra's address book can be flooded or taken over #1869](https://github.com/ZcashFoundation/zebra/issues/1869)
+- [Zebra does not evict pre-upgrade peers from the peer set across a network upgrade #706](https://github.com/ZcashFoundation/zebra/issues/706)
+- [Zebra accepts non-minimal height encodings #2226](https://github.com/ZcashFoundation/zebra/issues/2226)
+-  [Zebra nodes continually try to contact peers that always fail #1865](https://github.com/ZcashFoundation/zebra/issues/1865)
```
Comment (jvff) [inline README.md]: Minor: formatting typo

```suggestion
- [Zebra nodes continually try to contact peers that always fail #1865](https://github.com/ZcashFoundation/zebra/issues/1865)
```

**Review by jvff — DISMISSED** [review]

File: README.md [hunk 3]
```diff
@@ -13,17 +13,51 @@ consensus-compatible implementation of the Zcash protocol, currently under
...
+
+Zebra's network stack is interoperable with zcashd.
+Zebra implements all the features required to reach Zcash network consensus.
+
+Currently, Zebra does not validate the following Zcash consensus rules:
+
+#### NU5
+- ZIP-155 - Parse addrv2 in Zebra
+- Full validation of Orchard transactions from NU5 onwards
+   - Check that at least one of enableSpendsOrchard or enableOutputsOrchard is set #2433
```
Comment (oxarbitrage) [inline README.md]: I think we can skip the issue number here as we are not doing it with any other rule.
```suggestion
   - Check that at least one of enableSpendsOrchard or enableOutputsOrchard is set
```

File: README.md [hunk 4]
```diff
@@ -13,17 +13,51 @@ consensus-compatible implementation of the Zcash protocol, currently under
...
-- implement the Zcash proof of work consensus rules, and
-- sync on Mainnet under excellent network conditions.
+The goals of the beta release series are for Zebra to act as a fully validating Canopy and NU5 node, except for:
 
-Currently, Zebra does not validate all the Zcash consensus rules. It may be
+- mempool transactions,
+- block subsidies,
+- transaction fees,
+- some undocumented rules derived from Bitcoin, and
+- some consensus rules removed before Canopy activation. (Zebra checkpoints on Canopy activation.)
```
Comment (oxarbitrage) [inline README.md]: I don't think we need a comma on each list item here.
```suggestion
- Mempool transactions
- Block subsidies
- Transaction fees
- Some undocumented rules derived from Bitcoin
- Some consensus rules removed before Canopy activation (Zebra checkpoints on Canopy activation)
```

Comment (conradoplg) [reply README.md]: Do you want to remove the comma in the first item too?

Comment (oxarbitrage) [reply README.md]: Sorry, fixed the suggestion.

**Review by conradoplg — DISMISSED** [review]

**Review by oxarbitrage — APPROVED** [review]

**Pull request merged**